components/feature/syncedtabs/src/test/java/mozilla/components/feature/syncedtabs/storage/SyncedTabsStorageTest.kt

```diff
@@ -229,4 +235,69 @@ class SyncedTabsStorageTest {
         whenever(accountManager.authenticatedAccount()).thenReturn(null)
         assertEquals(null, feature.syncClients())
     }
+
+    @Test
+    fun `tabs are only stored when loaded`() = runBlockingTest {
+        val store = BrowserStore(
+            BrowserState(
+                tabs = listOf(
+                    createTab(id = "tab1", url = "https://www.mozilla.org", lastAccess = 123L),
+                    createTab(id = "tab2", url = "https://www.foo.bar", lastAccess = 124L),
+                ),
+                selectedTabId = "tab1"
+            )
+        )
+        val feature = spy(
+            SyncedTabsStorage(
+                accountManager,
+                store,
+                tabsStorage
+            )
+        )
+        feature.start()
+
+        // Change a tab besides loading it
+        store.dispatch(ContentAction.UpdateProgressAction("tab1", 50)).joinBlocking()
```

Comment:
Shouldn't we verify that no other `store` calls were made after the init? The state changes from `UpdateProgressAction` are not observed from the `SyncedTabsStorage`, so we should assert that this change does not call `store` again.

Reply:
I thought verifying that the `store` call occurred exactly once would be good enough to indicate that. Is there something else you'd suggest? `Mockito.any` has a Kotlin compatibility bug, so I'm not able to do the following:
```
        verify(tabsStorage, times(1)).store(
            eq(listOf(
                Tab(history = listOf(TabEntry(title = "", url = "https://www.mozilla.org", iconUrl = null)), active = 0, lastUsed = 123L),
                Tab(history = listOf(TabEntry(title = "", url = "https://www.foo.bar", iconUrl = null)), active = 0, lastUsed = 124L)
            ))
        )
        verify(tabsStorage, never()).store(any())
```

I tried with the custom `any()` matcher defined in the project as well, but it doesn't seem to allow sequenced verification like Mockito normally does.

Reply:
I think what you're looking for in this test is `reset`; it isn't a recommended approach for regular testing, but it helps for this one:

```kotlin

        // Tabs are only stored when initial state is collected, since they are already loaded
        verify(tabsStorage, times(1)).store(
            listOf(
                Tab(history = listOf(TabEntry(title = "", url = "https://www.mozilla.org", iconUrl = null)), active = 0, lastUsed = 123L),
                Tab(history = listOf(TabEntry(title = "", url = "https://www.foo.bar", iconUrl = null)), active = 0, lastUsed = 124L)
            )
        )

        // reset the mock so there are no verifications or prior calls on the mock.
        reset(tabsStorage)

        // Change a tab besides loading it
        store.dispatch(ContentAction.UpdateProgressAction("tab1", 50)).joinBlocking()

        verify(tabsStorage, never()).store(any())
```

Reply:
TY for the added detail! That does seem to work